chatlearn/models/vllm_module_v2.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2024 Alibaba Group Holding Limited. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""VLLM module"""
+
+import asyncio
+import inspect
+import os
+import sys
+from typing import Optional
+
+import torch
+from transformers import AutoTokenizer
+from vllm import SamplingParams
+from vllm.config import EngineConfig
+from vllm.executor.ray_utils import RayWorkerWrapper
+from vllm.engine.async_llm_engine import AsyncLLMEngine
+from vllm.engine.arg_utils import AsyncEngineArgs
+from vllm.usage.usage_lib import UsageContext
+from vllm.utils import FlexibleArgumentParser
+
+from chatlearn.utils.global_vars import set_vllm_actors
+from .torch_module import TorchModule
+
+
+class VLLMModuleV2(TorchModule, RayWorkerWrapper):
+    """VLLMModule is the class for vLLM models.
+
+    Args
+    ----
+    name : str
+        model name
+    """
+
+    def __init__(self, *args, **kwargs):
+        # avoid overwrite methods
+        methods_class1 = {method[0] for method in inspect.getmembers(TorchModule, predicate=inspect.isfunction)}
+        methods_class2 = {method[0] for method in inspect.getmembers(RayWorkerWrapper, predicate=inspect.isfunction)}
+        common_methods = methods_class1.intersection(methods_class2)
+        # common method is '__init__'
+        assert len(common_methods) == 1
```

Comment:
Why assume that these two classes have only 1 method in common?

Reply:
this is to avoid overwrite methods which may trigger bug